GUIDELINES.md

```diff
@@ -0,0 +1,159 @@
+# Technical and working style and guidelines
+
+*Challenging this documentation is encouraged. Please set up an issue for challenges, additions or corrections. For explanations, please consult the `dev` Slack channel*
+
+In this document we set out some basic "tips" (either mandatory or encouraged) which should guide your way of working, regardless of the task.
+
+* [Foreword](#foreword)
+* [`python is not pandas`](#python-is-not-pandas)
+* [Design patterns](#design-patterns)
+* [Programming](#programming)
+* [Critical thinking](#critical-thinking)
+* [Naming conventions](#naming-conventions)
+* [Spaces and spacing](#spaces-and-spacing)
+* [Comments and docs](#comments-and-docs)
+
+## Foreword
+
+In advance, we recommend installing the autoformatter [black](https://pypi.org/project/black/) in your IDE. In future we will use `flake8` (or similar) for automatically checking our python codebases. 
+
+Hopefully the conventions laid out here are the easy and intuitive set of [pep8](https://www.python.org/dev/peps/pep-0008/). 
+
+Code reviewers: it is on you to ensure that this style guide has been followed: there are no points for being lenient, but there [non-redeemable] points for being opinionated! We should all feel pressured into making sure that our code meets an acceptable standard.
```

Comment:
Is this intentionally repreated? Also maybe not clear how this tallies up with "(either mandatory or encouraged)" from above.

Reply:
yeah probs not needed!